postgres-testing/src/main/java/org/folio/postgres/testing/PostgresTesterContainer.java

```diff
@@ -34,44 +44,143 @@ public PostgresTesterContainer() {
    * Start the container.
    */
   @Override
-  public void start(String database, String username, String password) {
-    if (postgreSQLContainer != null) {
+  public void start(String database, String username, String password) throws PostgresTesterStartException {
+    if (primary != null) {
       throw new IllegalStateException("already started");
     }
-    postgreSQLContainer = new PostgreSQLContainer<>(dockerImageName)
-        .withDatabaseName(database)
-        .withUsername(username)
-        .withPassword(password)
-        .withStartupTimeout(Duration.ofSeconds(60));
-    postgreSQLContainer.start();
+
+    GenericContainer<?> tempStandby;
+    var replicationUser = "replicator";
+    var primaryHost = "primaryhost";
+    var dataDirectory = "/var/lib/postgresql/standby/";
+    var tempDirectory = "/tmp/standby/";
+    var hostVolume = "/tmp/rmb-standby-" + UUID.randomUUID();
+    var network = Network.newNetwork();
+
+    // In postgres replication the default is for commit to be async. So no configuration is needed, but our
+    // default for tests is to configure it for synchronous commit.
+    boolean configureSynchronousCommit = System.getProperty(POSTGRES_ASYNC_COMMIT) == null;
```

Comment:
I hear what you're saying, but think handling this with a system property is a good approach for now. Yes it could be handled differently, by making it part of the constructor or properties on the class, but then we would need to rewrite a lot of test code, which will take time that we don't have right now, without providing a lot of immediate value.

What I do think we need right now is a simple way to run all of the tests in either sync or async mode, and this gives us that.

Here's how I'm imagining this system property could be used. Say a team is having trouble with the read/write split in their code and they have traced this to one of RMB's methods. Say the problem is because replication is async in their prod/test env (which is very likely going to be the case). They can now verify and replicate this problem locally by setting this flag, running their test, and potentially submit a PR that they know will fix it because it can be replicated and tested locally.

The point is that right now (with this PR) we don't need to fix every problem that comes with async replication, but this property gives teams a simple way to solve them.

Another way this property could be used (eventually) is with automation. Jenkins could do two passes of `mvn test` to test all of the tests with either configuration or perhaps other configurations that emerge that we'd like to test. I say _eventually_ because, right now not every test passes with async. In my testing two tests intermittently fail for example. Is that a problem? I say no not today. Let's get this out there so teams can use it and decide how important it is to fix every test later. We can always open new tickets.